File: email/src/main/java/com/university/email/model/email/Email.java
package com.university.email.model.email;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.ArrayList;
import java.util.Date;
import java.util.LinkedList;
import java.util.Queue;

@Getter
@NoArgsConstructor
public class Email implements Comparable<Email> {
    private String sender;
    private Queue<String> receiver=new LinkedList<>();
    private String subject;
    private String body;
    private Object attachment;
    private Integer priority;
    private boolean readStatus;
    private Date date;

    public Email(EmailBuilder builder){
        this.sender=builder.getSender();
        this.receiver.addAll(builder.getReceiver());
        this.subject=builder.getSubject();
        this.body=builder.getBody();
        this.attachment=builder.getAttachment();
        this.priority=Integer.valueOf(builder.getPriority());
        this.readStatus=builder.isReadStatus();
        this.date=builder.getDate();
    }
    public void print(){
        System.out.println("");
        System.out.printf("Sender: %s%n", sender);
        ArrayList<String> receiversArr=new ArrayList<>(getReceiver());
        for(int i=0;i< receiversArr.size();i++)
            System.out.printf("Receiver %d: %s%n", i,receiversArr.get(i));
        System.out.printf("Subject: %s%n", subject);
        System.out.printf("Body: %s%n", body);
        System.out.printf("Priority: %s%n", priority);
    }
    @Override
    public int compareTo(Email email){
        return this.getPriority().compareTo(email.getPriority());
    }


}

File: email-front/src/app/mail/email.ts
export class Email {
  private sender: string;
  private receiver: Array<string>;
  private subject: string;
  private body: string;
  private date: Date;
  private readStatus: boolean;
  private priority: number;
  private attachment:FormData;

  constructor(sender: string, receiver: Array<string>, subject: string, body: string, priority: number, draft: boolean,attachment:FormData) {
    this.sender = sender;
    this.receiver = receiver;
    this.subject = subject;
    this.body = body;
    this.date = new Date();
    this.readStatus = false;
    this.priority = priority;
    this.attachment=attachment;
  }
  static createEmailFromObject(obj: any){
    return new Email(obj.sender, obj.receiver, obj.subject, obj.body, obj.priority, obj.readStatus,obj.attachment);
  }
  getFrom(){
    return this.sender;
  }
  getTo(){
    return this.receiver;
  }
  getSubject(){
    return this.subject;
  }
  getBody(){
    return this.body;
  }
  getDate(){
    return this.date.getDate() + '/' + (this.date.getMonth() + 1 ) + '/' + this.date.getFullYear();
  }
  getPriority(){
    return this.priority;
  }
  getRead(){
    return this.readStatus;
  }
  toggleRead(){
    this.readStatus = !this.readStatus;
  }
}
